File: Java/src/main/java/com/hamaraBasket/QualityChecker.java
package com.hamaraBasket;   

public class QualityChecker
    {
        public  void updateQuality(HamaraBasketItem item)
        {
            if (item.getSellBy() <= 0 && item.getQuality() > 1)
            {
                int quality = item.getQuality();
                item.setQuality(quality-2);
                return;
            }
            if (item.getQuality() == 0)
            {
                return;
            }
            int quality = item.getQuality();
            item.setQuality(quality - 1);
        }
    }

File: CSharpCore/HamaraBasketQualityAssurance/QualityChecker.cs
namespace HamaraBasketQualityAssurance
{
    public class QualityChecker
    {
        public virtual void UpdateQuality(HamaraBasketItem item)
        {
            if (item.SellBy <= 0 && item.Quality > 1)
            {
                item.Quality -= 2;
                return;
            }
            if (item.Quality == 0)
            {
                return;
            }
            item.Quality -= 1;
        }
    }
}
